### PR TITLE
Task conversions (Cats-Effect 1.0.0-RC3 update, part 2)

### PR DESCRIPTION
**Pull request by alexandru**
This is a continuation of PR #685 by @oleg-py ... I forked that PR and continued it.

Changes:

1. `Task.start` now forks, with `fork` being deprecated, done by @oleg-py in #685 
2. changed conversions to/from `Task` across the board, including their signatures
  - introduced for this purpose `TaskLike` and `TaskLift` type classes; changed `Observable` to use them instead of `Effect` because usage of `Effect` was wrong
3. made `unsafeCreate` and `Task.Context` private, because a major internals refactoring is planned and it's better to cleanup the API right now, those standing in the way
4. introduced `SafeApp` and deprecated `TaskApp`

### CODE REVIEW

File: monix-eval/shared/src/main/scala/monix/eval/TaskLike.scala [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+        Task.fromConcurrent(fa)
+    }
+}
+
+private[eval] abstract class ToTaskImplicits1 {
+  /**
+    * Converts to `Task` from
+    * [[https://typelevel.org/cats-effect/typeclasses/concurrent-effect.html cats.effect.Async]].
+    */
+  implicit def fromAsync[F[_]](implicit F: Effect[F]): TaskLike[F] =
```
Comment (oleg-py) [inline monix-eval/shared/src/main/scala/monix/eval/TaskLike.scala]: I know it's internal that won't be written explicitly, but maybe it should be named `fromEffect`? 😄

File: monix-eval/shared/src/main/scala/monix/eval/TaskLift.scala [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+/**
+  * A lawless type class that specifies conversions from `Task`
+  * to similar data types (i.e. pure, asynchronous, preferably
+  * cancelable).
+  */
+@implicitNotFound("""Cannot find implicit value for TaskLift[${F}].
+Building this implicit value might depend on having an implicit
+s.c.ExecutionContext in scope, a Scheduler or some equivalent type.""")
+trait TaskLift[F[_]] {
```
[the remark this answers is not in the document]
Comment (alexandru) [reply monix-eval/shared/src/main/scala/monix/eval/TaskLift.scala]: Meh, I like having `Task` prefix on the Task-related classes 🙂

File: monix-eval/shared/src/main/scala/monix/eval/SafeApp.scala [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  *   }
+  * }}}
+  *
+  * N.B. this is homologous with
+  * [[https://typelevel.org/cats-effect/datatypes/ioapp.html cats.effect.IOApp]],
+  * but meant for usage with [[Task]].
+  *
+  * Works on top of JavaScript as well ;-)
+  */
+trait SafeApp {
```
[the remark this answers is not in the document]
Comment (alexandru) [reply monix-eval/shared/src/main/scala/monix/eval/SafeApp.scala]: I renamed it to TaskApp.

**Comment by alexandru** [comment]
@oleg-py thanks a lot for the review, it was very useful and I addressed your comments.

**Pull request merged**